project/apps/identity_provider/models.py

```diff
@@ -36,22 +47,7 @@ class ApiKey(models.Model):
 
     last_modified = models.DateTimeField(blank=True, null=True)
 
-    def save(self, *args, **kwargs):
-        if (
-            not self.pk
-            and ApiKey.objects.filter(user=self.user, revoked=False).exists()
-        ):
-            # we might create new ApiKey but user already has one valid
-            logger.info(
-                f"User {self.user.username} already has active API Key. Skipping"
-            )
-            # get is intentional. exception here is mistake in logic
-            api_key = ApiKey.objects.get(user=self.user, revoked=False)
-            self.pk = api_key.pk
-            self.key = api_key.key
-            # this is to force update if possible
-            kwargs.update({"force_insert": False})
-            super(ApiKey, self).save(*args, **kwargs)
-        else:
-            # pk exists, then we are in update, we want to save. or user doesnt have valid keys
-            super(ApiKey, self).save(*args, **kwargs)
+    # Set an expiration date for the API key
+    expiry = models.DateTimeField(blank=False, null=False, default=default_expiration_date)
+
+    scope = models.CharField(max_length=50, blank=False, null=False, choices=scopeChoices, default='resource')
```

Comment:
If the scope have a choice, i think is better to not give a default, but instead who save the key should define the scope of it